examples/DictionaryExample/Program.cs

```diff
@@ -0,0 +1,194 @@
+        var setResponse = await client.DictionarySetAsync(cacheName: cacheName, dictionaryName: dictionaryName,
+            field: "my-field", value: "my-value", refreshTtl: false, ttlSeconds: 60);
+        if (setResponse is CacheDictionarySetResponse.Error setError)
+        {
+            _logger.LogInformation($"Error setting a value in a dictionary: {setError.Message}");
+            Environment.Exit(1);
+        }
+
+        // Set multiple values
+        var setBatchResponse = await client.DictionarySetBatchAsync(
+            cacheName: cacheName,
+            dictionaryName: dictionaryName,
+            new Dictionary<string, string>() {
+                { "field1", "value1" },
+                { "field2", "value2" },
+                { "field3", "value3" }},
+            refreshTtl: false);
+        if (setBatchResponse is CacheDictionarySetBatchResponse.Error setBatchError)
+        {
+            _logger.LogInformation($"Error setting a values in a dictionary: {setBatchError.Message}");
+            Environment.Exit(1);
+        }
+
+        // Get a value
+        var field = "field1";
+        var getResponse = await client.DictionaryGetAsync(
+            cacheName: cacheName,
+            dictionaryName: dictionaryName,
+            field: field);
+
+        var status = "";
+        var value = "";
+
+        if (getResponse is CacheDictionaryGetResponse.Hit unaryHit)
+        {
+            status = "HIT";
+            value = unaryHit.String();
+        }
+        else if (getResponse is CacheDictionaryGetResponse.Miss)
+        {
+            // In this example you can get here if you change the field name, or
+            // if you set a short TTL, then add a Task.Delay so that it expires.
+            status = "MISS";
+            value = "<NONE; operation was a MISS>";
+        }
+        else if (getResponse is CacheDictionaryGetResponse.Error getError)
+        {
+            _logger.LogInformation($"Error getting value from a dictionary: {getError.Message}");
+            Environment.Exit(1);
+        }
+
+        _logger.LogInformation("");
+        _logger.LogInformation($"Dictionary get of {field}: status={status}; value={value}");
+
+        // Get multiple values
+        var batchFieldList = new string[] { "field1", "field2", "field3", "field4" };
+        var getBatchResponse = await client.DictionaryGetBatchAsync(
+            cacheName: cacheName,
+            dictionaryName: dictionaryName,
+            fields: batchFieldList);
+        if (getBatchResponse is CacheDictionaryGetBatchResponse.Error getBatchError)
+        {
+            _logger.LogInformation($"Error getting value from a dictionary: {getBatchError.Message}");
+            Environment.Exit(1);
+        }
+
+        _logger.LogInformation("");
+        _logger.LogInformation("Displaying the result of dictionary get batch:");
+        var responses = ((CacheDictionaryGetBatchResponse.Success)getBatchResponse).Responses;
```

Comment:
could we move this up into the `if` statement, either before or after the error branch, so that we can use the pattern matching instead of the cast?